### ISSUE TITLE
Add Spleeter for source separation

### ISSUE THREAD

**Issue by shoegazerstella** [SwagLyrics/autosynch#30]
Hi,
I think it could be cool to add [spleeter](https://github.com/deezer/spleeter) for performing the source separation.
If you are interested in accepting a pull request I've done it [here](https://github.com/shoegazerstella/autosynch).

**Comment by aadibajpai** [comment]
Yes please! We have wanted to do that for quite some time but haven't really gotten around to it yet :(

**Comment by shoegazerstella** [comment]
Made this branch [https://github.com/shoegazerstella/autosynch/tree/spleeter-dl](https://github.com/shoegazerstella/autosynch/tree/spleeter-dl) in which I added spleeter and also the possibility to download the audio using [youtube-dl](https://github.com/ytdl-org/youtube-dl).

usage is below in the readme
```
python playback.py [artist] [song_title] -v spleeter
```
I also added a Dockerfile (but need to test properly), since I've got an error on macos when using open-unmix and wanted to test on ubuntu. That's the error:

```
RuntimeError: fft: ATen not compiled with MKL support
```

Made a lot of changes in the imports too, maybe I'll just create a pull request to add the new branch

**Comment by aadibajpai** [comment]
> Made this branch https://github.com/shoegazerstella/autosynch/tree/spleeter-dl in which I added spleeter and also the possibility to download the audio using [youtube-dl](https://github.com/ytdl-org/youtube-dl).
> 
> usage is below in the readme
> 
> ```
> python playback.py [artist] [song_title] -v spleeter
> ```
> 
> I also added a Dockerfile (but need to test properly), since I've got an error on macos when using open-unmix and wanted to test on ubuntu. That's the error:
> 
> ```
> RuntimeError: fft: ATen not compiled with MKL support
> ```
> 
> Made a lot of changes in the imports too, maybe I'll just create a pull request to add the new branch

Yes a new branch would be great!

So I thought about integrating youtube-dl but then the issue was sometimes the music video has additional audio before or after the actual song like https://www.youtube.com/watch?v=DK_0jXPuIr0 that can throw off sync.

That said, it should work for most songs so that is very welcome too 😄

**Comment by chriswang030** [comment]
youtube-dl would be great but as Aadi mentioned the biggest trouble is finding a suitable video without too much wacky stuff going on. You'd be better off downloading the file yourself.

**Comment by aadibajpai** [comment]
A metric I was thinking of was if the duration is the same on both Spotify and YouTube then the audio can be expected to be the same too. Yet to test it though, and it probably involves Spotify API usage

**Comment by DumbMachine** [comment]
![image](https://user-images.githubusercontent.com/23381512/75600799-e1262d00-5ad9-11ea-9963-d7fc9b7bc5d5.png)
`Genius API`, many times gives the time stamp where the music begins in a video. Though this may still have some errors as in many `music videos` there are segments without music in between. 
A `hacky` way of using youtube-dl would be to search for music with tags `lyrics`, `audio` etc.
